src/biotite/sequence/profile.py

```diff
@@ -0,0 +1,165 @@
+# This source code is part of the Biotite package and is distributed
+# under the 3-Clause BSD License. Please see 'LICENSE.rst' for further
+# information.
+
+import numpy as np
+from .seqtypes import NucleotideSequence, ProteinSequence
+from .alphabet import LetterAlphabet
+
+__name__ = "biotite.sequence"
+__author__ = "Maximilian Greil"
+_all__ = ["SequenceProfile", "to_consensus"]
+
+# Abbreviations
+_NUC_DNA_ALPH = NucleotideSequence.alphabet_unamb
+_NUC_RNA_ALPH = LetterAlphabet(["A", "C", "G", "U"])
+_PROT_ALPH = ProteinSequence.alphabet
+
+
+class SequenceProfile(object):
+    """
+    A :class:`SequenceProfile` object stores information about a sequence
+    profile of aligned sequences, i.e. the consensus sequence.
+
+    This class saves the position frequency matrix of the occurrences of
+    each alphabet symbol at each position. It also saves the number of gaps
+    at each position.
+
+    A :class:`SequenceProfile` object can be created from an indefinite number
+    of aligned sequences.
+
+    All attributes of this class are publicly accessible.
+
+    Parameters
+    ----------
+    position_frequency_matrix : ndarray, dtype=int, shape=(n,k)
+        This matrix simply saves for each position how often absolutely each symbol is present.
+    position_gaps : ndarray, dtype=int, shape=n
+        Array which indicates the number of gaps at each position.
+    alphabet : Alphabet, length=k
+        Alphabet of sequences of sequence profile
+
+    Attributes
+    ----------
+    position_frequency_matrix : ndarray, dtype=int, shape=(n,k)
+        This matrix simply saves for each position how often absolutely each symbol is present.
+    position_gaps : ndarray, dtype=int, shape=n
+        Array which indicates the number of gaps at each position.
+    alphabet : Alphabet, length=k
+        Alphabet of sequences of sequence profile
+    """
+
+    def __init__(self, position_frequency_matrix, position_gaps, alphabet):
+        self.position_frequency_matrix = position_frequency_matrix
+        self.position_gaps = position_gaps
+        self.alphabet = alphabet
```

Comment:
I suggest that the public access is handled via `@property`. For example I think the user should not be able to change the alphabet, because then the size of the frequency matrix does not fit anymore. Instead I think `position_frequency_matrix` and `position_frequency_matrix` should check whether the shape is still the same, when these attributes are set.

Furthermore, the variable names `position_frequency_matrix` and `position_frequency_matrix` are quite long in my opinion. May I suggest `symbols` and `gaps` instead?